test/helpers/components/oneagent/daemonset.go

```diff
@@ -29,9 +45,36 @@ func Get(ctx context.Context, resource *resources.Resources, dynakube dynatracev
 	}).Get()
 }
 
-func ForEachPod(ctx context.Context, resource *resources.Resources, dynakube dynatracev1beta1.DynaKube, actionFunc daemonset.PodConsumer) error {
-	return daemonset.NewQuery(ctx, resource, client.ObjectKey{
-		Name:      dynakube.OneAgentDaemonsetName(),
-		Namespace: dynakube.Namespace,
-	}).ForEachPod(actionFunc)
+func CreateUninstallDaemonSet(namespace string) features.Func {
+	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
+		oaDaemonSet := manifests.ObjectFromFile[*appsv1.DaemonSet](t, uninstallOneAgentDaemonSetPath)
+		oaDaemonSet.Namespace = namespace
```

Comment:
should we also set the `Tolerations` for the daemonset according to the `Tolerations` set in the dynakube?

Reply:
yes good idea, otherwise it could happen that agent runs on nodes where cleanup would not be run afterwards